### PR TITLE
Add support for SSL_(read/peek/write)_ex

### PR DESCRIPTION
**Pull request by samuel40791765**
### Issues:
Resolves `CryptoAlg-1852`

### Description of changes: 
mySQL consumes the `SSL_(read/peek/write)_ex` family of APIs from OpenSSL1.1.1. These are similar to the existing `SSL_(read/peek/write)` APIs in AWS-LC, but have slightly different expected parameters and function signatures.
Since these have the same underlying operations, I just wrapped `SSL_*_ex` around it's respective `SSL_*` operation.

* More information can be found in OpenSSL's documentation here: https://www.openssl.org/docs/man1.1.1/man3/SSL_read_ex.html

### Call-outs:
N/A

### Testing:
I extended an existing test suite that was testing the `SSL_pending` and `SSL_(read/peek/write)`, and created one that used `SSL_(read/peek/write)_ex` instead.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: include/openssl/ssl.h [hunk 1]
```diff
@@ -372,9 +372,27 @@ OPENSSL_EXPORT int SSL_accept(SSL *ssl);
 // https://crbug.com/466303.
 OPENSSL_EXPORT int SSL_read(SSL *ssl, void *buf, int num);
 
+// SSL_read_ex reads up to |num| bytes from |ssl| into |buf|. It is similar to
+// |SSL_read|, but instead of returning the number of bytes read, it returns
+// 1 on success or 0 for failure. The number bytes actually read is stored in
```
Comment (dkostic) [inline include/openssl/ssl.h:377]:
```suggestion
// 1 on success or 0 for failure. The number of bytes actually read is stored in
```

File: include/openssl/ssl.h [hunk 2]
```diff
@@ -372,9 +372,27 @@ OPENSSL_EXPORT int SSL_accept(SSL *ssl);
...
+// This is only maintained for OpenSSL compatibility. Use |SSL_read| instead.
+OPENSSL_EXPORT int SSL_read_ex(SSL *ssl, void *buf, size_t num,
+                               size_t *read_bytes);
+
 // SSL_peek behaves like |SSL_read| but does not consume any bytes returned.
 OPENSSL_EXPORT int SSL_peek(SSL *ssl, void *buf, int num);
 
+// SSL_peek_ex reads up to |num| bytes from |ssl| into |buf|. It is similar to
+// |SSL_peek|, but instead of returning the number of bytes read, it returns
+// 1 on success or 0 for failure. The number bytes actually read is stored in
```
Comment (dkostic) [inline include/openssl/ssl.h:389]:
```suggestion
// 1 on success or 0 for failure. The number of bytes actually read is stored in
```

**Review by dkostic — DISMISSED** [review]

File: ssl/ssl_test.cc [hunk 3]
```diff
@@ -5552,6 +5552,52 @@ TEST_P(SSLVersionTest, SSLPending) {
...
+  UniquePtr<SSL> ssl(SSL_new(client_ctx_.get()));
+  ASSERT_TRUE(ssl);
+  EXPECT_EQ(0, SSL_pending(ssl.get()));
+
+  ASSERT_TRUE(Connect());
+  EXPECT_EQ(0, SSL_pending(client_.get()));
+  EXPECT_EQ(0, SSL_has_pending(client_.get()));
+
+  size_t buf_len;
+  ASSERT_TRUE(SSL_write_ex(server_.get(), "hello", 5, &buf_len));
```
Comment (andrewhop) [inline ssl/ssl_test.cc:5567]:
```suggestion
  ASSERT_EQUAL(SSL_write_ex(server_.get(), "hello", 5, &buf_len), 1);
```
Would return 5 still count as "true" and pass this test? You do that down below for the read/peek tests.

Comment (samuel40791765) [reply ssl/ssl_test.cc]: Yeah my bad, I was planning on this to be consistent.

**Review by dkostic — APPROVED** [review]

**Pull request merged**